mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -739,6 +739,15 @@ impl MakerSwap {
         };
         swap_events.push(MakerSwapEvent::MakerPaymentInstructionsReceived(instructions));
 
+        let taker_amount = MmNumber::from(self.taker_amount.clone());
```

Comment:
`MmNumber::from` only takes a reference inside the function. You can implement:  

```rs
impl From<&BigDecimal> for MmNumber {
    fn from(n: &BigDecimal) -> MmNumber { from_dec_to_ratio(n).into() }
}
```  

to avoid unnecessary cloning in cases like this.

Reply:
yeah, we have such issue
https://github.com/KomodoPlatform/komodo-defi-framework/issues/2338

Reply:
@borngraced could you please introduce `From<&BigDecimal> for MmNumber` and start using it for your changes. after pr merge someone can take refactoring as a separate task.

Reply:
done